libcxx/include/__iterator/iterator_traits.h

```diff
@@ -46,6 +46,9 @@
 
 _LIBCPP_BEGIN_NAMESPACE_STD
 
+template <class _Iter>
+using __iter_reference_t _LIBCPP_NODEBUG = decltype(*std::declval<_Iter&>());
```

Comment:
I would do consistently with the other typedefs and use `iterator_traits<>::reference` instead.